Source/Scene/GltfLoaderUtil.js

```diff
@@ -0,0 +1,188 @@
+import Check from "../Core/Check.js";
+import defaultValue from "../Core/defaultValue.js";
+import defined from "../Core/defined.js";
+import Sampler from "../Renderer/Sampler.js";
+import TextureMagnificationFilter from "../Renderer/TextureMagnificationFilter.js";
+import TextureMinificationFilter from "../Renderer/TextureMinificationFilter.js";
+import TextureWrap from "../Renderer/TextureWrap.js";
+
+/**
+ * glTF loading utilities.
+ *
+ * @namespace GltfLoaderUtil
+ *
+ * @private
+ */
+var GltfLoaderUtil = {};
+
+/**
+ * Get the uri or buffer view for an image.
+ *
+ * @param {Object} options Object with the following properties:
+ * @param {Object} options.gltf The glTF JSON.
+ * @param {Number} options.imageId The image ID.
+ * @param {SupportedImageFormats} options.supportedImageFormats The supported image formats.
+ *
+ * @returns {Object} An object containing a <code>uri</code> and <code>bufferView</code> property.
+ */
+GltfLoaderUtil.getImageUriOrBufferView = function (options) {
+  options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+  var gltf = options.gltf;
+  var imageId = options.imageId;
+  var supportedImageFormats = options.supportedImageFormats;
+
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.object("options.gltf", gltf);
+  Check.typeOf.number("options.imageId", imageId);
+  Check.typeOf.object("options.supportedImageFormats", supportedImageFormats);
+  //>>includeEnd('debug');
+
+  var image = gltf.images[imageId];
+  var extras = image.extras;
+
+  var bufferViewId = image.bufferView;
+  var uri = image.uri;
+
+  // First check for a compressed texture
+  if (defined(extras) && defined(extras.compressedImage3DTiles)) {
```

Comment:
I kept support for compressed textures like KTX, Crunch, and `compressedImage3DTiles` in this PR but didn't unit test any of that functionality because it's going to be remove soon (CC https://github.com/CesiumGS/cesium/issues/9479)